pkg/controllers/user/systemimage/systemimage.go

```diff
@@ -0,0 +1,142 @@
+package systemimage
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/rancher/pkg/controllers/user/workload"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	systemImageVersionAnn = "field.cattle.io/systemImageVersion"
+	systemProjectAnn      = "management.cattle.io/system-namespace"
+)
+
+var systemProjectLabels = labels.Set(map[string]string{"authz.management.cattle.io/system-project": "true"})
+
+type Syncer struct {
+	clusterName        string
+	projectLister      v3.ProjectLister
+	projects           v3.ProjectInterface
+	workloadController workload.CommonController
+}
+
+func (s *Syncer) Sync(key string, obj *v3.Project) error {
+	projects, err := s.projectLister.List(s.clusterName, systemProjectLabels.AsSelector())
```

Comment:
This is done for every project sync, I wonder if we can check at the start and go forward only when we find that `obj is system project`.(and probably avoid the loop for finding system project)

Reply:
I used to do that way, but I find system project may not update for quite a long time, so the system images will not update until the system project updated.